### PR TITLE
Improve the output of the healthcheck service

### PR DESCRIPTION
**Pull request by ydrisR**
- Change the output of the service: it now returns the full list of services even when one is down.
- Return more details comming from the services themselves

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        
+        {
```
Comment (CedricPerotto) [inline .vscode/launch.json:7]: This launch.json configuration works for me : 
```suggestion:-0+11
        {
            "type": "node",
            "request": "launch",
            "name": "Launch Program",
            "program": "${workspaceFolder}/index.js",
            "env": {
                "SERVICE_PORT":"9180",
                "MONITORED_URLS":"[{\"name\": \"gatekeeper\", \"url\":\"http://localhost:9123/status\"},{\"name\": \"hakken\", \"url\":\"http://localhost:8000/status\"},{\"name\": \"highwater\", \"url\":\"http://localhost:9191/status\"},{\"name\": \"hydrophone\", \"url\":\"http://localhost:9157/status\"},{\"name\": \"message-api\", \"url\":\"http://localhost:9119/status\"},{\"name\": \"platform auth\", \"url\":\"http://localhost:9222/status\"},{\"name\": \"platform user\", \"url\":\"http://localhost:9221/status\"},{\"name\": \"platform data\", \"url\":\"http://localhost:9220/status\"},{\"name\": \"platform notification\", \"url\":\"http://localhost:9223/status\"},{\"name\": \"platform task\", \"url\":\"http://localhost:9224/status\"},{\"name\": \"seagull\", \"url\":\"http://localhost:9120/status\"},{\"name\": \"shoreline\", \"url\":\"http://localhost:9107/status\"},{\"name\": \"tide-whisperer\", \"url\":\"http://localhost:9127/status\"},{\"name\": \"portal-api\", \"url\":\"http://localhost:9507/status\"},{\"name\": \"confirm-email\", \"url\":\"http://localhost:9508/status\"}]"
            }
        }
```

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            "request": "attach",
+            "name": "Attach",
+            "port": 9229
+        },
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Launch Program",
+            "program": "${workspaceFolder}/app.js"
+        }
```
Comment (franck-fourel) [inline .vscode/launch.json:36]: Would be great to have a real launch.json file. 
Here is the one I used locally:

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "10.13.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Launch Health-check",
            "program": "${workspaceFolder}\\index.js",
            "env": {
                "SERVICE_PORT":"9190",
                "MONITORED_URLS":"[{\"name\": \"shoreline\", \"url\":\"http://localhost:9107/status\"},{\"name\": \"gatekeeper\", \"url\":\"http://localhost:9123/status\"},{\"name\": \"hydrophone\", \"url\":\"http://localhost:9157/status\"}, {\"name\": \"platform-data\", \"url\":\"http://localhost:9220/status\"}]"
            }
        }
    ]
}
```

**Review by CedricPerotto — CHANGES_REQUESTED** [review]
Ok for me. There is only the launch.json added in the PR that doesn't work for me (either delete it or check my suggestion).

File: service.js [hunk 3]
```diff
@@ -8,98 +8,134 @@ var restify = require('restify');
...
-        this.services=services;    
+        this.services = services;
         //default logger
         this.logger = bunyan.createLogger({
             name: 'health-check-default',
             streams: [
                 {
-                  level: 'info',
-                  stream: process.stdout
+                    level: 'info',
```
Comment (franck-fourel) [inline service.js:27]: Not sure why I cannot change this level of logging. 
Even if I change the LOG_LEVEL environment variable

File: test/test_service.js [hunk 4]
```diff
@@ -25,29 +25,51 @@ describe('Global health check service', () => {
...
+                res.body.length.should.eql(3);
                 done();
             });
         });
+        it('should return an error for shoreline', () => {           
+            let authStatus = statusResults.filter(status => status.service == "shoreline");
+            authStatus[0].status.should.eql("NOK");
+            authStatus[0].error.should.eql("500 - \"cannot reach server\"");
+        })
     });
```
Comment (franck-fourel) [inline test/test_service.js]: would be great to add test cases for :
- return 200 but no body
- return 200 with malformed JSON body
- return 200 with JSON body
basically what was added in this PR

**Review by franck-fourel — COMMENTED** [review]
couple of minor things we can discuss

File: test/test_service.js (hunk 4 above)
Comment (ydrisR) [reply test/test_service.js]: Done ;)

File: service.js (hunk 3 above)
Comment (ydrisR) [reply service.js:27]: There was a problem with the environment variable case: loglevel instead of logLevel...

**Review by CedricPerotto — APPROVED** [review]

**Review by franck-fourel — APPROVED** [review]
LGTM

**Pull request merged**